utils/config_generator.py

```diff
@@ -0,0 +1,392 @@
+import argparse
+import re
+import subprocess
+import sys
+from io import StringIO
+from pathlib import Path
+
+import ruamel.yaml
+
+from utils.find_corpus import (
+    fetch_mtdata,
+    fetch_news_crawl,
+    fetch_opus,
+    fetch_sacrebleu,
+    get_remote_file_size,
+)
+
+"""
+Generate a training config for a language pair based on the latest production
+training config, taskcluster/configs/config.prod.yml.
+"""
+
+root_dir = Path(__file__).parent.parent
+prod_config_path = root_dir / "taskcluster/configs/config.prod.yml"
+
+pretrained_student_models = {
+    ("ru", "en"): "https://storage.googleapis.com/releng-translations-dev/models/ru-en/better-teacher/student"
+}  # fmt: skip
+
+skip_datasets = [
+    # The NLLB dataset is based off of the CCMatrix dataset, and is mostly duplicated.
+    "CCMatrix",
+    # Skip Multi* datasets as they are generally multilingual versions of the original datasets.
+    "MultiMaCoCu",
+    "MultiHPLT",
+    # In Russian, the WikiTitles data had its direction reversed. The `LinguaTools-WikiTitles`
+    # version is fine.
+    "WikiTitles",
+]
+
+# Do not include small datasets. This works around #508, and minimizes dataset tasks that
+# won't bring a lot more data.
+minimum_dataset_sentences = 200
+
+flores_101_languages = {
+    "af", "amh", "ar", "as", "ast", "az", "be", "bn", "bs", "bg", "ca", "ceb", "cs", "ckb", "cy",
+    "da", "de", "el", "en", "et", "fa", "fi", "fr", "ful", "ga", "gl", "gu", "ha", "he", "hi",
+    "hr", "hu", "hy", "ig", "id", "is", "it", "jv", "ja", "kam", "kn", "ka", "kk", "kea", "km",
+    "ky", "ko", "lo", "lv", "ln", "lt", "lb", "lg", "luo", "ml", "mr", "mk", "mt", "mn", "mi",
+    "ms", "my", "nl", "nb", "npi", "nso", "ny", "oc", "om", "or", "pa", "pl", "pt", "pus", "ro",
+    "ru", "sk", "sl", "sna", "snd", "so", "es", "sr", "sv", "sw", "ta", "te", "tg", "tl", "th",
+    "tr", "uk", "umb", "ur", "uz", "vi", "wo", "xh", "yo", "zh", "zh", "zu"
+}  # fmt: skip
+
+
+def get_git_revision_hash() -> str:
+    return subprocess.check_output(["git", "rev-parse", "HEAD"]).decode("ascii").strip()
+
+
+def update_config(prod_config: any, name: str, source: str, target: str) -> dict[str, str]:
+    experiment = prod_config["experiment"]
+
+    # Update the prod config for this language pair.
+    experiment["name"] = name
+    experiment["src"] = source
+    experiment["trg"] = target
+    experiment["bicleaner"]["dataset-thresholds"] = {}
+
+    pretrained_model = pretrained_student_models.get((source, target))
+    if pretrained_model:
+        experiment["teacher-mode"] = "two-stage"
+        experiment["pretrained-models"]["train-backwards"]["urls"] = [pretrained_model]
+    else:
+        experiment["pretrained-models"] = {}
+
```

Comment:
There should be also this here:
```
        experiment["teacher-mode"] = "one-stage"
```

But I found another use case when we don't want that. If the amount of mono-trg data is too small (for example for en-lt) we still want to use two-stage. We don't want to loop over 5M back-translated sentences.

Reply:
Let's follow-up with rules for this, or manually configure the configs after they are generated.